src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/ElementsMarshalling.cs

```diff
@@ -107,40 +167,74 @@ public BlittableElementsMarshalling(TypeSyntax managedElementType, TypeSyntax un
 
         public override StatementSyntax GenerateUnmanagedToManagedByValueOutMarshalStatement(TypePositionInfo info, StubCodeContext context)
         {
-            // MemoryMarshal.CreateSpan(ref MemoryMarshal.GetReference(<GetUnmanagedValuesSource>), <GetUnmanagedValuesSource>.Length)
-            ExpressionSyntax destination = CastToManagedIfNecessary(
-                InvocationExpression(
-                    MemberAccessExpression(
-                        SyntaxKind.SimpleMemberAccessExpression,
-                        ParseName(TypeNames.System_Runtime_InteropServices_MemoryMarshal),
-                        IdentifierName("CreateSpan")),
-                    ArgumentList(
-                        SeparatedList(new[]
-                        {
-                            Argument(
-                                InvocationExpression(
-                                    MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression,
-                                        ParseName(TypeNames.System_Runtime_InteropServices_MemoryMarshal),
-                                        IdentifierName("GetReference")),
-                                    ArgumentList(SingletonSeparatedList(
-                                        Argument(CollectionSource.GetUnmanagedValuesSource(info, context))))))
-                                .WithRefKindKeyword(
-                                    Token(SyntaxKind.RefKeyword)),
-                            Argument(
-                                MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression,
-                                    CollectionSource.GetUnmanagedValuesSource(info, context),
-                                    IdentifierName("Length")))
-                        }))));
-
-            // <GetManagedValuesDestination>.CopyTo(<source>);
-            return ExpressionStatement(
+            ExpressionSyntax destination;
+            List<StatementSyntax> statements = new();
+            var numElementsAssignment = CollectionSource.GetNumElementsAssignmentFromManagedValuesSource(info, context);
+            statements.Add(numElementsAssignment);
+            if (MarshallerHelpers.MarshalsOutToLocal(info, context))
+            {
+                // #pragma warning disable CS9081
```

Comment:
We shouldn't be suppressing this rule. This suppression likely covers up a use-out-of-scope bug. We should either declare `native_out` as `scoped`, or (if we can't), revisit how we do this. Can you share some example code that goes down this route?